src/lava/proc/rf/models.py

```diff
@@ -0,0 +1,121 @@
+import numpy as np
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.ports import PyInPort, PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+from lava.proc.rf.process import RF
+from copy import deepcopy
+
+class AbstractPyRFModelFloat(PyLoihiProcessModel):
+    a_real_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, float)
+    a_imag_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, float)
+    s_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, bool, precision=1)
+    real: np.ndarray = LavaPyType(np.ndarray, float)
+    imag: np.ndarray = LavaPyType(np.ndarray, float)
+    sin_decay: float = LavaPyType(float, float)
+    cos_decay: float = LavaPyType(float, float)
+    vth: float = LavaPyType(float, float)
+
+    def sub_thresh_dynamics(self):
+
+        a_real_in_data = self.a_real_in.recv()
+        a_imag_in_data = self.a_imag_in.recv()
+
+        old_real = deepcopy(self.real)
```

Comment:
Is `deepcopy` really necessary here?

Reply:
Yeah, I think I went a bit overboard with the deepcopies there